mlflow/pyfunc/__init__.py

```diff
@@ -940,6 +945,14 @@ def load_model(
             f'Model does not have the "{FLAVOR_NAME}" flavor',
             RESOURCE_DOES_NOT_EXIST,
         )
+    # TODO: improve this logic if we start allowing code with custom loader modules
+    if conf.get(_MODEL_CODE_PATH) is not None and "pyfunc" in conf.get("loader_module"):
+        flavor_code_path = conf.get(_MODEL_CODE_PATH)
+        code_path = os.path.join(
+            local_path,
+            os.path.basename(flavor_code_path),
+        )
+        return _load_model_code_path(code_path)
```

Comment:
Is there a reason why we want to return here and skip all the following things like warning for incompatible py version, adding code from conf to system path, etc.?

Reply:
moved down